tests/certification/bindings/azure/storagequeues/storagequeue_test.go

```diff
@@ -0,0 +1,409 @@
+
+			req := &daprClient.InvokeBindingRequest{Name: "standard-binding-2", Operation: "create", Data: []byte(msg), Metadata: metadata}
+			err := client.InvokeOutputBinding(ctx, req)
+			require.NoError(ctx, err, "error publishing message")
+		}
+
+		// Assertion on the data.
+		messagesFor1.Assert(ctx, time.Minute)
+		messagesFor2.Assert(ctx, time.Minute)
+
+		return nil
+	}
+
+	application := func(ctx flow.Context, s common.Service) (err error) {
+		// Setup the input binding endpoints.
+		err = multierr.Combine(err,
+			s.AddBindingInvocationHandler("standard-binding-1", func(_ context.Context, in *common.BindingEvent) ([]byte, error) {
+				messagesFor1.Observe(string(in.Data))
+				ctx.Logf("Got message: %s", string(in.Data))
+				return []byte("{}"), nil
+			}),
+			s.AddBindingInvocationHandler("standard-binding-2", func(_ context.Context, in *common.BindingEvent) ([]byte, error) {
+				messagesFor2.Observe(string(in.Data))
+				ctx.Logf("Got message: %s", string(in.Data))
+				return []byte("{}"), nil
+			}))
+		return err
+	}
+
+	flow.New(t, "servicebusqueue certification").
+		// Run the application logic above.
+		Step(app.Run("standardApp", fmt.Sprintf(":%d", appPort), application)).
+		Step(sidecar.Run("standardSidecar",
+			embedded.WithAppProtocol(runtime.HTTPProtocol, appPort),
+			embedded.WithDaprGRPCPort(grpcPort),
+			embedded.WithDaprHTTPPort(httpPort),
+			embedded.WithComponentsPath("./components/standard"),
+			runtime.WithOutputBindings(
+				binding_loader.NewOutput("azure.storagequeues", func() bindings.OutputBinding {
+					return binding_asq.NewAzureStorageQueues(log)
+				}),
+			),
+			runtime.WithInputBindings(
+				binding_loader.NewInput("azure.storagequeues", func() bindings.InputBinding {
+					return binding_asq.NewAzureStorageQueues(log)
+				}),
+			),
+			runtime.WithSecretStores(
+				secretstores_loader.New("local.env", func() secretstores.SecretStore {
+					return secretstore_env.NewEnvSecretStore(log)
+				}),
+			))).
+		Step("interrupt network", network.InterruptNetwork(time.Minute, []string{}, []string{}, "5671", "5672")).
+		Step("send and wait", test).
+		Run()
+}
+
+func TestAzureStorageQueueTTLs(t *testing.T) {
+	log := logger.NewLogger("dapr-components")
+	ttlMessages := watcher.NewUnordered()
+
+	ports, _ := dapr_testing.GetFreePorts(3)
+	grpcPort := ports[0]
+	httpPort := ports[1]
+	appPort := ports[2]
+
+	ttlTest := func(ctx flow.Context) error {
+		client, err := daprClient.NewClientWithPort(fmt.Sprintf("%d", grpcPort))
+		require.NoError(t, err, "Could not initialize dapr client.")
+
+		ctx.Logf("Sending messages for expiration.")
+		for i := 0; i < numOfMessages; i++ {
+			msg := fmt.Sprintf("Expiring message %d", i)
+
+			metadata := make(map[string]string)
+
+			// Send message with TTL set in yaml file
+			messageTTLReq := &daprClient.InvokeBindingRequest{Name: "ttl-binding", Operation: "create", Data: []byte(msg), Metadata: metadata}
+			messageTTLReq.Metadata["ttlInSeconds"] = "10"
```

Comment:
Please test the TTL works without including this at the message level, and please test that using a different TTL at the message level overwrites the default component level TTL.

Example: no request metadata TTL -> verify 10 second TTL applied
Message TTL set to 15 seconds -> verify 15 second TTL applied